fem/transfer.hpp

```diff
@@ -22,6 +22,228 @@
 namespace mfem
 {
 
+/** @brief Base class for transfer algorithms that construct transfer Operator%s
+    between two finite element (FE) spaces. */
+/** Generally, the two FE spaces (domain and range) can be defined on different
+    meshes. */
+class GridTransfer
+{
+protected:
+   FiniteElementSpace &dom_fes; ///< Domain FE space
+   FiniteElementSpace &ran_fes; ///< Range FE space
+
+   /** @brief Desired Operator::Type for the construction of all operators
+       defined by the underlying transfer algorithm. It can be ignored by
+       derived classes. */
+   Operator::Type oper_type;
+
+   OperatorHandle fw_t_oper; ///< Forward true-dof operator
+   OperatorHandle bw_t_oper; ///< Backward true-dof operator
+
+#ifdef MFEM_USE_MPI
+   bool parallel;
+#endif
+   bool Parallel() const
+   {
+#ifndef MFEM_USE_MPI
+      return false;
+#else
+      return parallel;
+#endif
+   }
+
+   const Operator &MakeTrueOperator(FiniteElementSpace &fes_in,
+                                    FiniteElementSpace &fes_out,
+                                    const Operator &oper,
+                                    OperatorHandle &t_oper);
+
+public:
+   /** Construct a transfer algorithm between the domain, @a dom_fes_, and
+       range, @a ran_fes_, FE spaces. */
+   GridTransfer(FiniteElementSpace &dom_fes_, FiniteElementSpace &ran_fes_);
+
+   /// Virtual destructor
+   virtual ~GridTransfer() { }
+
+   /** @brief Set the desired Operator::Type for the construction of all
+       operators defined by the underlying transfer algorithm. */
+   /** The default value is Operator::ANY_TYPE which typically corresponds to a
+       matrix-free operator representation. Note that derived classes are not
+       required to support this setting and can ignore it. */
+   void SetOperatorType(Operator::Type type) { oper_type = type; }
+
+   /** @brief Return an Operator that transfers GridFunction%s from the domain
+       FE space to GridFunction%s in the range FE space. */
+   virtual const Operator &ForwardOperator() = 0;
+
+   /** @brief Return an Operator that transfers GridFunction%s from the range FE
+       space back to GridFunction%s in the domain FE space. */
+   virtual const Operator &BackwardOperator() = 0;
+
+   /** @brief Return an Operator that transfers true-dof Vector%s from the
```

Comment:
Minor comments, maybe its the editor being used, but in the documentation I see the percentage symbol (%) instead of ' , for example on line 83 Vector%s  instead of Vector's.

Reply:
I think this is Doxygen syntax: `Vector` will create an auto-link to the `Vector` class, but `Vectors` won't (because there is no `Vectors` class). `Vector%s` will create an auto-link to the `Vector` class followed by an unlinked `s` (see e.g. [here](http://mfem.github.io/doxygen/html/classmfem_1_1GridTransfer.html) in the Doxygen docs).